core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -535,7 +535,7 @@ private RelationPlan getInsertPlan(
                 if (supportsMissingColumnsOnInsert) {
                     continue;
                 }
-                expression = new Constant(column.getType(), null);
+                expression = new Constant(column.getType(), column.getDefaultValue().orElse(null));
```

Comment:
If the connector `supportsMissingColumnsOnInsert`, then it is supposed to apply the default value on its own, right?
Do we have tests to cover both cases (connector applies the default value versus the engine applies it here)?

Reply:
Right. We don't have such a test. I'm wondering if we can write a helpful test covering both cases in the engine.